src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -95,50 +62,16 @@ const MyDiagnoses = ({ navigation }) => {
     />
   )
 
-  useEffect(
-    () => {
-      AnalyticsService.setCurrentScreenName('My Diagnoses')
-      const unsuscribe = DatabaseService.fetchSolvedDiagnosesFromCurrentUser(async (snapshot) => {
-        setDownloadingDiagnoses(true)
-        snapshot.docChanges().map(docChange => {
-          const doc = docChange.doc
-          const changeType = docChange.type
-
-          switch (changeType) {
-            case 'added': setClosedDiagnoses(v => processAddedDiagnose(v, doc)); break
-            case 'modified': setClosedDiagnoses(v => processModifiedDiagnose(v, doc)); break
-            case 'removed': setClosedDiagnoses(v => processRemovedDiagnose(v, doc)); break
-          }
-        })
-        setDownloadingDiagnoses(false)
-      })
-      return () => { unsuscribe() }
-    },
-    []
-  )
-
-  useEffect(
-    () => {
-      const unsuscribe = DatabaseService.fetchUnsolvedDiagnosesFromCurrentUser(async (snapshot) => {
-        setDownloadingDiagnoses(true)
-        snapshot.docChanges().map(docChange => {
-          const doc = docChange.doc
-          const changeType = docChange.type
-
-          switch (changeType) {
-            case 'added': setOpenDiagnoses(v => processAddedDiagnose(v, doc)); break
-            case 'modified': setOpenDiagnoses(v => processModifiedDiagnose(v, doc)); break
-            case 'removed': setOpenDiagnoses(v => processRemovedDiagnose(v, doc)); break
-          }
-        })
-        setDownloadingDiagnoses(false)
-      })
-      return () => { unsuscribe() }
-    },
-    []
-  )
+  const { results: openDiagnoses, loading: openDiagnosesLoading } = useQueryListener({
+    fetch: DatabaseService.fetchUnsolvedDiagnosesFromCurrentUser
+  })
+  const { results: closedDiagnoses, loading: closedDiagnosesLoading } = useQueryListener({
+    fetch: DatabaseService.fetchSolvedDiagnosesFromCurrentUser
+  })
+  const downloadingDiagnoses = openDiagnosesLoading || closedDiagnosesLoading
 
   useEffect(() => {
+    AnalyticsService.setCurrentScreenName('My Diagnoses')
```

Comment:
This line cannot be in this `useEffect`. It'll be executed whenever any of the dependencies for this hook changes. Move it to its own effect, like this:
```javascript
useEffect(() => {
  AnalyticsService.setCurrentScreenName('My Diagnoses')
}, [])
```